File: backend/app/schemas/smart_universe.py
from pydantic import BaseModel


class SmartTraderOut(BaseModel):
    """
    Public representation of a smart trader entry used by universe-related APIs.

    Most fields come directly from `smart_trader_universe`, combined with the
    human-readable `address` from the `traders` table.
    """

    trader_id: int
    address: str
    window_days: int

    score: float
    win_rate_window: float
    pnl_window: float
    volatility_window: float
    max_drawdown_window: float
    payoff_ratio: float
    expectancy: float
    trades_per_day: float

File: frontend/src/api/smartUniverse.ts
import { api } from "./client";

export interface SmartTrader {
  trader_id: number;
  address: string;
  window_days: number;
  
  score: number;
  win_rate_window: number;
  pnl_window: number;
  volatility_window: number;
  max_drawdown_window: number;
  payoff_ratio: number;
  expectancy: number;
  trades_per_day: number;
}

export interface FetchSmartUniverseParams {
  window_days?: number;
  min_score?: number;
  min_trades_per_day?: number;
}

export async function fetchSmartUniverse(params?: FetchSmartUniverseParams): Promise<SmartTrader[]> {
  const res = await api.get<SmartTrader[]>("/smart-universe", { params });
  return res.data;
}

